### PR TITLE
refac: basis data

### PR DESCRIPTION
**Pull request by johevemi**
Refactor code so child metering points are not saved as basis data when only energy results are run

### CODE REVIEW

File: source/databricks/calculation_engine/package/calculation/calculation.py [hunk 1]
```diff
@@ -94,8 +94,10 @@ def _execute(
         )
     )
 
-    metering_point_periods = get_metering_points_and_child_metering_points(
-        metering_point_periods_df
+    metering_point_periods = all_metering_point_periods.filter(
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/package/calculation/calculation.py:97]: Perhaps we can use a more specific variable name to counter future problems?

File: source/databricks/calculation_engine/package/calculation/calculation.py [hunk 2]
```diff
@@ -94,8 +94,10 @@ def _execute(
...
+        (f.col(Colname.metering_point_type) == MeteringPointType.CONSUMPTION.value)
+        | (f.col(Colname.metering_point_type) == MeteringPointType.PRODUCTION.value)
+        | (f.col(Colname.metering_point_type) == MeteringPointType.EXCHANGE.value)
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/package/calculation/calculation.py:100]: Move to a function?

File: source/databricks/calculation_engine/package/calculation/wholesale/get_metering_points_and_child_metering_points.py [hunk 3]
```diff
@@ -21,25 +21,24 @@
 
 
 @logging_configuration.use_span("get_metering_points_and_child_metering_points")
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/package/calculation/wholesale/get_metering_points_and_child_metering_points.py:23]: Update this text?

File: source/databricks/calculation_engine/tests/calculation/wholesale/test_get_metering_points_and_child_metering_points.py
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/tests/calculation/wholesale/test_get_metering_points_and_child_metering_points.py:1]: Rename test file?

Comment (johevemi) [reply source/databricks/calculation_engine/tests/calculation/wholesale/test_get_metering_points_and_child_metering_points.py:1]: ah yes, it only returns child_metering_points now, I will rename it to get_child_metering_points

File: source/databricks/calculation_engine/package/calculation/wholesale/get_metering_points_and_child_metering_points.py (hunk 3 above)
Comment (johevemi) [reply source/databricks/calculation_engine/package/calculation/wholesale/get_metering_points_and_child_metering_points.py]: yes

File: source/databricks/calculation_engine/package/calculation/calculation.py [hunk 4]
```diff
@@ -122,14 +132,31 @@ def _execute(
...
+        results.basis_data = basis_data_factory.create(
+            args,
+            metering_points_periods_for_energy_calculation,
+            metering_point_time_series,
+        )
 
     return results
 
 
+def get_metering_points_periods_for_energy_calculation(
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/package/calculation/calculation.py:150]: Thx. It, however, is better suited outside this calculations module, which should focus on orchestration and not details.
E.g. see the [Single Level of Abstraction Principle](https://medium.com/@outfunnel/the-single-level-of-abstraction-principle-52b5d56ef54b).

File: source/databricks/calculation_engine/package/calculation/calculation.py (hunk 1 above)
Comment (johevemi) [reply source/databricks/calculation_engine/package/calculation/calculation.py]: I have made some changes that should give a bit more readability, please take a look an see if you agree

File: source/databricks/calculation_engine/package/calculation/calculation.py (hunk 4 above)
Comment (johevemi) [reply source/databricks/calculation_engine/package/calculation/calculation.py:150]: okay, I will move the function into get_child_metering_points and refactor the the file back to what it was and move out of wholesale folder

File: source/databricks/calculation_engine/package/calculation/calculation.py [hunk 5]
```diff
@@ -122,10 +126,17 @@ def _execute(
...
+        results.basis_data = basis_data_factory.create(
+            args, metering_point_periods_for_wholesale, metering_point_time_series
+        )
+    else:
+        # Add basis data to results
+        results.basis_data = basis_data_factory.create(
+            args,
+            metering_points_periods_for_energy_calculation,
+            metering_point_time_series,
+        )
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/package/calculation/calculation.py:139]: I anticipate that SME or PO has confirmed this difference in behavior.

Comment (BjarkeMeier) [reply source/databricks/calculation_engine/package/calculation/calculation.py:139]: @ClausQAEnerginet will you include tests of this difference in behavior?

File: source/databricks/calculation_engine/package/calculation/preparation/transformations/metering_point_periods_for_calculation_type.py [hunk 6]
```diff
@@ -20,26 +20,53 @@
...
+    Returns all child metering points.
     The energy supplier of child metering points is added from its parent metering point.
     """
-    production_consumption_and_exchange_metering_points = (
-        _get_production_consumption_and_exchange_metering_points(
-            metering_point_periods_df
-        )
+    production_and_consumption_metering_points = all_metering_point_periods.filter(
+        (f.col(Colname.metering_point_type) == MeteringPointType.CONSUMPTION.value)
+        | (f.col(Colname.metering_point_type) == MeteringPointType.PRODUCTION.value)
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/package/calculation/preparation/transformations/metering_point_periods_for_calculation_type.py:61]: I anticipate that exchange can't have child metering points(?)

Comment (johevemi) [reply source/databricks/calculation_engine/package/calculation/preparation/transformations/metering_point_periods_for_calculation_type.py:61]: correct

File: source/databricks/calculation_engine/package/calculation/calculation.py [hunk 7]
```diff
@@ -121,10 +123,16 @@ def _execute(
             args,
             prepared_charges,
         )
+    else:
+        metering_point_periods_for_basis_data = (
+            get_metering_point_periods_for_energy_basis_data(all_metering_point_periods)
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/package/calculation/calculation.py:128]: Are there any - or plans about - tests that child MP have energy supplier?

**Review by BjarkeMeier — APPROVED** [review]

File: source/databricks/calculation_engine/package/calculation/preparation/transformations/metering_point_periods_for_calculation_type.py [hunk 8]
```diff
@@ -20,26 +20,71 @@
...
+    Returns all metering point periods that should be included in an energy and wholesale calculations basis data.
+    All child metering points are updated with the energy supplier id of its parent metering point.
+    """
+    non_child_metering_points_periods = all_metering_point_periods.filter(
+        (f.col(Colname.metering_point_type) == MeteringPointType.CONSUMPTION.value)
+        | (f.col(Colname.metering_point_type) == MeteringPointType.PRODUCTION.value)
+        | (f.col(Colname.metering_point_type) == MeteringPointType.EXCHANGE.value)
+    )
+    child_metering_points_with_energy_suppliers = (
+        _get_child_metering_points_with_energy_suppliers(all_metering_point_periods)
```
[the remark this answers is not in the document]
Comment (johevemi) [reply source/databricks/calculation_engine/package/calculation/preparation/transformations/metering_point_periods_for_calculation_type.py:49]: well yes, that would solve a lot the issues here

**Pull request merged**